server/tls.go

```diff
@@ -0,0 +1,81 @@
+package server
```

Comment:
I'm not 100% happy with this sitting in the `server` package, because of importing in `storage` package. I'm open to ideas. `server/tls` maybe? Or simply `tls`. But it may be confusing with Go standard library.